### DISCUSSION TITLE
[Feature] Automatic details for flight, based on flight number

### DISCUSSION THREAD

**Discussion by yurividal** [johanohly/AirTrail#44] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi. It would we awesome if we could use something like [aviationstack](https://aviationstack.com/) free API to automatically retrieve all the details about a flight, given the flight number.

**Comment by johanohly** [comment]
Sadly have not found a free API that can find a flight from just the flight number and date

**Comment by yurividal** [comment]
I believe aviationstack , which i have linked above,  allows for this. You would just have to add a field in the config for the user to put their own API key, and the free tier should be enough for small, self-hosted instances.

**Comment by johanohly** [comment]
Historical data is limited to the basic plan and above. For example `https://api.aviationstack.com/v1/flights?access_key=ACCESS_KEY&flight_date=2023-09-21&flight_iata=SK1585` yields a `Your current subscription plan does not support this API function.` error.

**Comment by yurividal** [comment]
ohh, thats unfortunate. i guess i didn't read the fine-prints. does that also apply for flights in the future?

**Comment by johanohly** [comment]
No - it seems like any use of the `flight_date` parameter is restricted.

**Comment by johanohly** [comment]
I have implemented a basic version of this feature and it will be available in the next release. 
Using https://adsbdb.com you can now retrieve (and autofill) the origin and destination airport, and the airline.

**Reply by johanohly** [reply]
Added in v0.3.0
